Sponsor banners

mdlincoln opened the pull request:
Closes #1563

In this example, I've put "alert" boxes at both the top and bottom of lesson pages so we can see if we like one or both. I have also mocked up example formatting for the alert boxes in lessons, so we could accommodate longer or shorter texts, call-out headers, etc.

See an example lesson here: https://deploy-preview-1585--ph-preview.netlify.com/

todo:

- [x] move the `.sitewide-alert` scoped css to the main CSS file once finalized
- [x] move sitewide banner to an include? it only shows up once in the layouts so does not necessarily warrant being an include, but doing so would follow the same pattern as the support-alert.html box
- [x] finalize text for both alerts
- [x] ES translation for both texts
- [x] FR translation for both texts

mdlincoln committed:
Create sponsorship alerts on lesson pages

mdlincoln committed:
add sitewide sponsor banner

mdlincoln committed:
document support-alert include

acrymble commented:
Thanks @mdlincoln. Are these easy to turn on / off, and which bits are changeable for a given message?

drjwbaker commented:
Thanks @mdlincoln. Simultaneously subtle and in your face. Quite a design achievement :)

mdlincoln commented:
@acrymble see the documentation for turning these on/off in the _config.yml file

The text itself is editable in the _snippets.yml file.

Next step is for @programminghistorian/development-team to push their desired text to this PR.

mdlincoln commented:
Reminder to @programminghistorian/development-team to add the desired text to this PR so that we can start the translation process.

acrymble commented:
How about this @drjwbaker ?:

Top: Great Open Access tutorials cost money to produce. Join [link] the growing number of people supporting Programming Historian so we can continue to share knowledge without boundaries [ link ]

Bottom: If you found this lesson helpful, support us[link to patreon] so that we can bring you more high quality open access tutorials.

Please change as you see appropriate.

drjwbaker commented:
So sorry. This PR slipped through my (manual) filter. 

@acrymble Reads well, except 'without boundaries', which seems odd to me. Perhaps:

`Top: Great Open Access tutorials cost money to produce. Join [link] the growing number of people supporting Programming Historian so we can continue to share knowledge free of charge[/link]`

acrymble commented:
I know it's a bit awkward @mdlincoln, but if possible can we just roll this out first in English, check on the response, and then ask for translation? I suspect we'll want to tinker with it a few times, and I hate asking for multiple translations or being stuck with what we first think of. I can do the subsequent pull request.

mdlincoln commented:
Are you proposing

1. having the banners up only on English pages? or
2. having the banners up on all pages, but only with English text?

either could work, but I'd code it differently. Also, especially if you do 2 I think the @programminghistorian/spanish-team  and @programminghistorian/french-team  teams should weigh in

jenniferisasi commented:
My take on this is that although the English site receives more traffic than the ES and FR do, combined, our users navigate both the ES/FR and EN versions more often than not. I think having banners in the EN version but not the ES/FR versions could be a bit confusing for the users and send different messages. 

Top: Producir buenos tutoriales de acceso abierto cuesta dinero. Únete [link] al creciente número de personas que apoyan a Programming Historian para que podamos continuar compartiendo conocimientos de forma gratuita [link]

Bottom: Si esta lección te ha resultado útil, [link]apóyanos[link] para que podamos seguir produciendo tutoriales en abiertos y de gran calidad.

acrymble commented:
Ok @jenniferisasi if you think it's important, I respect that. Can we decouple these from "translations" and instead have "multilingual messages" that can be changed independently to appeal relevant to different linguistic audiences? We don't really need exact translations to be effective here, and it would dramatically simplify our workflows for advertising campaigns.

drjwbaker commented:
Hard agree @acrymble. I really like sponsorship messages becoming something that each language team owns and tweaks to their audiences. As Sponsorship Manager, I'm happy to manage divergence.

mdlincoln committed:
move sitewide banner to an include

mdlincoln committed:
move scoped CSS into main CSS file

mdlincoln committed:
add sitewide banner to lang-home + lesson pages

They use the blank.html template instead, so weren't covered
by the base.html template edits

jenniferisasi commented:
I don't see any problem with the proposed "new" way. There are differences in perception, (dis)advantages and need for OA materials in the contexts of our audience anyways. It also looks like it is just a matter of adding the different messages in the snippets file, by the looks of the files that @mdlincoln is editing, and that is easy.

drjwbaker commented:
@JMParr As @programminghistorian/global-team lead, do you have any thoughts on this idea? (in sum: sponsorships banners different in each language edition so as to target "their" readers)

JMParr commented:
> @JMParr As @programminghistorian/global-team lead, do you have any thoughts on this idea? (in sum: sponsorships banners different in each language edition so as to target "their" readers)

Hi, @drjwbaker - I'm inclined to agree with @jenniferisasi 's observations here. One of the challenges we've seen with multilingual DH is that English has frequently remained the default instructional language. Having sponsorship banners in each language is one step to address this problem.

acrymble commented:
I agree that we should have banners in each language. But I'm concerned that our attempts always to do what's right means we can't ever test something out quickly and simply. No business rolls out a service in every market at the same time. They test things, refine them, and roll out in stages. I admit to sometimes feeling paralyzed by our desire to always translate everything before seeing if it works, because when it doesn't, nobody wants to do the work again to improve it.

drjwbaker commented:
@acrymble The *specific* difference here from normal practice is that we are proposing that the EN, ES and FR banners are different, that they diverge to target their audiences/needs/campaigns. Sorry, @JMParr, I didn't make that clear in my summary!

> I agree that we should have banners in each language. But I'm concerned that our attempts always to do what's right means we can't ever test something out quickly and simply. No business rolls out a service in every market at the same time. They test things, refine them, and roll out in stages. I admit to sometimes feeling paralyzed by our desire to always translate everything before seeing if it works, because when it doesn't, nobody wants to do the work again to improve it.

Sorry you feel this way @acrymble. This feels like a separate discussion to have on a different ticket, with examples of the unwillingness to improve that you mention. Note that in this case the delay in getting the banners done had nothing to do with language editors providing text, and everything to do with me deleting the email notification for https://github.com/programminghistorian/jekyll/pull/1585#issuecomment-564542648 in error.

acrymble commented:
@drjwbaker you're right. Conversation for elsewhere.

drjwbaker commented:
what is outstanding here?

mdlincoln commented:
We need @programminghistorian/spanish-team  and @programminghistorian/french-team content for the snippets at `donation-alert` (the text appearing on each lesson) and `donation-banner` (the short text appearing at the top of every page)

drjwbaker commented:
Thanks for confirming @mdlincoln. I was concerned it was no longer clear after the chatter that lead to #1620.

jenniferisasi commented:
Unless @rivaquiroga jumps in quick on this one, for now and to get it out promptly, we can use my translation above, copied here: 

Top: Producir buenos tutoriales de acceso abierto cuesta dinero. Únete [link] al creciente número de personas que apoyan a Programming Historian para que podamos continuar compartiendo conocimientos de forma gratuita [link]

Bottom: Si esta lección te ha resultado útil, [link]apóyanos[link] para que podamos seguir produciendo tutoriales en abiertos y de gran calidad.

spapastamkou commented:
Top:
Les tutoriels de qualité à accès libre ont un coût de production. Merci de
rejoindre [link] celles et ceux qui soutiennent le Programming Historian et
contribuent au libre partage des savoirs.
Bottom:
Si cette leçon vous a été utile, merci de nous soutenir [link to patreon]
pour continuer à mettre en ligne de nouveaux tutoriels à accès libre.

Is that all or should we contribute in this commit as
well? https://github.com/programminghistorian/jekyll/pull/1585/commits/ba68285c1f148656135b9e1bd9481931668a06b6

mdlincoln commented:
Yes, please add these content to the _snippets.yml file on this branch, then I think we'll be ready to merge.

spapastamkou committed:
test FR and EN texts

spapastamkou committed:
Update snippets.yml

spapastamkou committed:
Update snippets.yml

adds text in FR

spapastamkou commented:
@mdlincoln I was not very clear between the texts circulating in comment and the ones in the commit. I made a test with FR and EN texts so that you can catch easily the changes and see if they correspond to what you had in mind. 
In short, I translated the donation-banner text as it was in the commit; and replaced the donation-alert existing text with what is qualified as top text in the comments. 
Not sure what to do with the bottom text (in comments); and did not touch the ES part to avoid mistakes. 
Please check and let me know.

jenniferisasi committed:
Update snippets.yml

Commit:
Update snippets.yml

fixed broken links to Patreon

drjwbaker commented on _data/snippets.yml:
```diff
@@ -378,6 +378,39 @@ donate:
+
+    ---
+
+    More text here as well
```
Can we remove this?

drjwbaker commented on _data/snippets.yml:
```diff
@@ -378,6 +378,39 @@ donate:
+
+    ---
+
+    More text here as well
```
Can we remove this?

drjwbaker commented on _data/snippets.yml:
```diff
@@ -378,6 +378,39 @@ donate:
+
+    ---
+
+    More text here as well
```
Can we remove this?

drjwbaker commented:
@jenniferisasi @spapastamkou Thanks for you work on this. Looks like we have some text to get going with.

I've fixed some links and notes the `More text here as well` that probably should go. @acrymble: are you happy to proceed?

mdlincoln committed:
remove other standin donation alert text

drjwbaker reviewed (dismissed):
Perfect :)

spapastamkou committed:
Update snippets.yml

small correction FR part

drjwbaker reviewed (approved):
@spapastamkou I'll take your word for it that the FR version needed that edit :)

spapastamkou commented:
@drjwbaker yes, sorry, my own mistake:)
